src/flood_model/serve_model.py

```diff
@@ -461,13 +461,23 @@ def style_exists(style_name: str) -> bool:
     bool
         True if the style exists, although it may be empty.
         False if it does not exist.
+
+    Raises
+    -------
+    HTTPError
+        If geoserver responds with anything but OK or NOT_FOUND, raises it as an exception since it is unexpected.
     """
     response = requests.get(
         f'{get_geoserver_url()}/styles/{style_name}.sld',
         auth=(EnvVariable.GEOSERVER_ADMIN_NAME, EnvVariable.GEOSERVER_ADMIN_PASSWORD)
     )
-    response.raise_for_status()
-    return response.status_code == HTTPStatus.OK
+    if response.status_code == HTTPStatus.OK:
+        return True
+    if response.status_code == HTTPStatus.NOT_FOUND:
+        return False
+    # If it does not meet the expected results then raise an error
+    # Raise error manually, so we can configure the text
+    raise requests.HTTPError(response.text, response=response)
```

Comment:
That's not quite what I'm trying to do. Before this PR the behaiviour was that if the style was not found, it would raise an error. I want it to return False. Yours succeeds in fixing that but it also has the potential of masking any other errors that aren't Style Not Found